pylint/checkers/len.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2016 Alexander Todorov <atodorov@MrSenko.com>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+"""check for incorrect usage of len() in conditions
+"""
+import itertools
+from astroid.node_classes import Call, Const, Name, UnaryOp
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+MSGS = {
+    'W1801': ('Do not use `len(SEQUENCE)` as condition value',
+              'len-as-condition',
+              'Used when Pylint detects incorrect use of len(sequence) inside \
+              conditions.'),
+    }
+
+
+def _is_len_call(node):
+    """checks if @node is len(SOMETHING)"""
+    return isinstance(node, Call) and isinstance(node.func, Name) and node.func.name == 'len'
```

Comment:
Can you also ensure `node` refers to the built-in `len`?

Unfortunately there's two methods that implement this check in the codebase ☹️ 

https://github.com/PyCQA/pylint/blob/2beb5e650d689d4903676b5ca6d82a1806bbbe79/pylint/checkers/refactoring.py#L342-L346
https://github.com/PyCQA/pylint/blob/23e59c80cfcba5878b7f634464a203017ab91187/pylint/checkers/python3.py#L44-L45